transport/http2_server.go

```diff
@@ -683,20 +683,38 @@ func (t *http2Server) handleWindowUpdate(f *http2.WindowUpdateFrame) {
 	})
 }
 
+func createHeaderFieldsFromMD(headerFields []hpack.HeaderField, md metadata.MD) []hpack.HeaderField {
```

Comment:
Nit: maybe "appendHeaderFieldsFromMD"?  This modifies the slice and returns it just like `append` does.

Reply:
done.